base/src/ImageViewerModule.cpp

```diff
@@ -8,13 +8,73 @@
 #include "Logger.h"
 #include "Utils.h"
 
-class ImageViewerModule::Detail
+#if defined(__arm__) || defined(__aarch64__)
+#include "ApraNvEglRenderer.h"
+#include "DMAFDWrapper.h"
+#include "Command.h"
+#endif
+
+class DetailRenderer
 {
 
 public:
-	Detail(std::string& strTitle): mStrTitle(strTitle) {}
+	DetailRenderer(ImageViewerModuleProps& _props) : props(_props) {}
+
+	~DetailRenderer() 
+	{
+		#if defined(__arm__) || defined(__aarch64__)
+		 if(renderer)
+        {
+            delete renderer;
+        }
+		#endif
+	}
+
+	virtual bool compute() = 0;
+
+	bool eglInitializer(uint32_t _height, uint32_t _width)
+	{
+	#if defined(__arm__) || defined(__aarch64__)
+        uint32_t displayHeight, displayWidth;
+        NvEglRenderer::getDisplayResolution(displayWidth,displayHeight);
```

Comment:
do a sanity check for width < displayWidth and same for height. Make sure that the program does not crash

Reply:
added the check